### PR TITLE
Update action to support Node 20

### PR DESCRIPTION
**Pull request by neilstudd**
RE: [GitHub Announcement, March 2024](https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/) -

> Following on from our [warning in workflows using Node16](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/) we will start enforcing the use of Node20 rather than Node16 on the 3rd of June.

Currently, this action outputs a warning about its usage of Node 16, but starting from 3rd June 2024, this will become a failure at runtime.

I've forked my own version to ensure it continues running locally; this is what worked for me (it's almost all ChatGPT-generated as I'm out of my depth):

* Changed Dockerfile to specify a Node 20 base image
* The `RUN` command failed until ~I~ ChatGPT converted it to a Node 20 friendly version
* Presumably the Node 20 base image uses a different file structure, because this led to a runtime error: `fatal: detected dubious ownership in repository at '/github/workspace'` - added the necessary `safe.directory` into the script (and had to promote user name/email to global for it to pick them up, too)

Long story short, the old behaviour is restored with these changes, but you might prefer to make your own, better changes (or at least squash-merge to deal with all my random debugging steps)

### CODE REVIEW

**Comment by TheLastProject** [comment]
Thanks for trying to contribute! However, this MR seems to be made under the incorrect assumption that this action is causing the warning, which I do not believe is the case.

As is usual for ChatGPT (a system whose only goal is to generate "plausible sounding responses" and which should not be used to actually code with), its suggestion is complete bogus. This action is a very simple shell script of only a few lines (see `entrypoint.sh`, it's the whole codebase) which does not have any dependencies except coreutils, wget and git. Changing from an Alpine to a Node base image would have added over 375MB of unused cruft, increasing this action's base container's filesize ~125-fold (from ~3MB to ~380MB) for no reason.

Looking at another repo of mine that uses it, the actual GitHub warning message seems to be as follows:

> Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: **actions/checkout@v2**. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/. (emphasis mine).

The problem here is the `v2` release of `actions/checkout`. Luckily a new version of that was released ages ago so that instruction in the README can simply be updated.

I've updated the outdated README instruction in 4306447c72144c1ae10770722bb18220413c4625 and, while I was add it, updated the alpine base container too and added dependabot to help keep the Alpine container more up-to-date in the future. I've released this as `v1.0.1` (because the updates are useful for security), but the README change should be what fixes the actual warning.
